github/rules.go

```diff
@@ -0,0 +1,1208 @@
+// Copyright 2025 The go-github AUTHORS. All rights reserved.
```

Comment:
Since everything in this file related to rulesets, do you want to rename this file from "rules.go" to "rulesets.go"?
(And its companion, "rules_test.go" to "rulesets_test.go"?)

Reply:
I just realised I'd got the naming slightly wrong; `RepositoryRulesetRule` -> `RepositoryRule`. The _rules.go_ file contains the following types.

```mermaid
flowchart LR
    A[RepositoryRuleset] --> B[RepositoryRulesetRules] --> C[RepositoryRule]
    D[BranchRules] --> C
```

Reply:
Your mermaid diagram is really nice and yet so simple... Would GitHub display this diagram properly if it were included in a comment at the top of `rules.go` when viewing that file?
(Just curious - no action necessary.)

Reply:
So thoughts on if we need any more changes to this PR?

Reply:
> I just realised I'd got the naming slightly wrong; `RepositoryRulesetRule` -> `RepositoryRule`. The _rules.go_ file contains the following types.

I was holding off until you addressed this, but if you are good to go, I'll review it.

Reply:
Sorry I thought I'd pushed, I've pushed now.